Lib/asyncio/selector_events.py

```diff
@@ -386,6 +386,41 @@ def _sock_recv(self, fut, registered, sock, n):
         else:
             fut.set_result(data)
 
+    def sock_recv_into(self, sock, buf):
+        """Receive data from the socket.
+
+        The received data is written into *buf* (a writable buffer).
+        The return value is the number of bytes written.
+
+        This method is a coroutine.
+        """
+        if self._debug and sock.gettimeout() != 0:
+            raise ValueError("the socket must be non-blocking")
+        fut = self.create_future()
+        self._sock_recv_into(fut, False, sock, buf)
+        return fut
```

Comment:
If you want, you can further optimize this by transforming `sock_recv_into` into an `async def` coroutine, and creating the `fut` object only when `sock.recv_into(buf)` fails with a `BlockingError`.

Reply:
I don't think that's very useful, as people will mainly want to use that for large data. Right now the implementation mirrors that of `sock_recv`, which is easier to understand.